### PR TITLE
fix: update app settings for App Services

### LINKED ISSUE #906: [AVM Module Issue]: App Services (microsoft.web/sites) application settings are incorrectly constructed
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/web/site

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

the module automatically populate the following App Settings:

1. AzureWebJobsStorage
2. AzureWebJobsDashboard
3. APPINSIGHTS_INSTRUMENTATIONKEY
4. APPLICATIONINSIGHTS_CONNECTION_STRING

* According to the documentation [HERE](https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#appinsights_instrumentationkey), `APPINSIGHTS_INSTRUMENTATIONKEY` and `APPLICATIONINSIGHTS_CONNECTION_STRING` should not be used at the same time. when possible, use `APPLICATIONINSIGHTS_CONNECTION_STRING`.
* According to the documentation [HERE](https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#azurewebjobsstorage__accountname), `AzureWebJobsStorage` should only be used if the Shared Key Access is not disabled for the storage account. 
* According to the documentation [HERE](https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#azurewebjobsdashboard), `AzureWebJobsDashboard` is deprecated.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by tyconsulting**
## Description

closes #906 

- Added a new parameter `storageAccountUseIdentityAuthentication`
```
@description('Optional. If the provided storage account requires Identity based authentication (\'allowSharedKeyAccess\' is set to false). When set to true, the minimum role assignment required for the App Service Managed Identity to the storage account is \'Storage Blob Data Owner\'.')
param storageAccountUseIdentityAuthentication bool = false
```
I had to add this parameter because the storage account property `allowSharedKeyAccess` is not returned when looking up existing storage accounts. Therefore we have to use boolean parameter to determine which variable to create (`AzureWebJobsStorage` or `AzureWebJobsStorage__accountName`)

- Removed parameter `setAzureWebJobsDashboard` because it is now obsolete.

## Pipeline

[avm.res.web.site](https://github.com/tyconsulting/bicep-registry-modules/actions/workflows/avm.res.web.site.yml)

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [x] This is a bug fix:
  - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [x] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [x] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

**Comment by tyconsulting** [comment]
hey @tsc-buddy could you please review this PR when you get a chance? thanks

**Comment by tyconsulting** [comment]
Hey @AlexanderSehr could you please get someone to review this PR for me?

**Comment by tsc-buddy** [comment]
A review and deployment of the changes has been scheduled for 26/2/24 once I return back to work

**Comment by AlexanderSehr** [comment]
> A review and deployment of the changes has been scheduled for 26/2/24 once I return back to work

Thanks @tsc-buddy, this resouce sure sees a lot of love. Let us know if you need support.

File: avm/res/web/site/version.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-    "$schema": "https://aka.ms/bicep-registry-module-version-file-schema#",
-    "version": "0.2",
-    "pathFilters": [
-        "./main.json"
-    ]
+  "$schema": "https://aka.ms/bicep-registry-module-version-file-schema#",
+  "version": "1.0",
```
Comment (ChrisSidebotham) [inline avm/res/web/site/version.json:3]: Please revert this to `0.3` until AVM Module are considered GA

Comment (ChrisSidebotham) [reply avm/res/web/site/version.json:3]: @tyconsulting

**Comment by tyconsulting** [comment]
@ChrisSidebotham version number reverted back. please review. thanks

**Comment by ChrisSidebotham** [comment]
Looks good to me! @tsc-buddy - Please give your review as the module owner! :D

**Review by tsc-buddy — DISMISSED** [review]
Thanks for your patience & updates @tyconsulting. Appreciate the support.

**Review by tsc-buddy — APPROVED** [review]
Version file conflict fixed

File: avm/res/web/site/config--appsettings/main.bicep [hunk 2]
```diff
@@ -18,23 +18,23 @@ param kind string
...
-  } : {})) : {}
+var azureWebJobsValues = !empty(storageAccountResourceId) && !(storageAccountUseIdentityAuthentication) ? {
+  AzureWebJobsStorage: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};AccountKey=${storageAccount.listKeys().keys[0].value};'
+} : !empty(storageAccountResourceId) && storageAccountUseIdentityAuthentication ? union(
+  { AzureWebJobsStorage__accountName: storageAccount.name },
+  { AzureWebJobsStorage__blobServiceUri: storageAccount.properties.primaryEndpoints.blob }
+) : {}
 
 var appInsightsValues = !empty(appInsightResourceId) ? {
-  APPINSIGHTS_INSTRUMENTATIONKEY: appInsight.properties.InstrumentationKey
```
Comment (krbar) [inline avm/res/web/site/config--appsettings/main.bicep:37]: @tyconsulting @tsc-buddy Was the removal of the `APPINSIGHTS_INSTRUMENTATIONKEY` intentional? I'm experiencing issues with the App Insights connection.

Comment (tyconsulting) [reply avm/res/web/site/config--appsettings/main.bicep:37]: hey @krbar the reason was stated in the linked issue #906 
https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#appinsights_instrumentationkey
image

**Pull request merged**